Add snap support

robgev opened the pull request:
---

**What's this PR do?**

Fixes #443: Adds snap support

**You have tested this PR on:**
  - [ ] Windows
  - [x] Linux/Ubuntu
  - [ ] macOS

robgev committed:
Build: Add snap support

This adds new build configs in package.json for electron-builder
to be able to generate snap package for zulip. Also, this will
change development.md line about the packages outputted.

Fixes #443.

akashnimare commented:
Few thoughts - 
* Have you tested it locally, does it build the snap installer? This works for me on 64-bit. Can you check if we can build 32-bit snap from a 64-bit machine?
* We also need to commit the generated `snapcraft.yaml` file, more info [here](https://docs.snapcraft.io/build-snaps/electron). I have registered `zulip` so you could use this name.
* Check out the Zulip commit message guidelines for more details: https://zulip.readthedocs.io/en/latest/version-control.html

robgev commented:
1. I checked 64 bit, don't know about 32, but I will try.
2. Sure thing, I will add snapcraft.yaml, but I don't exactly know where. Should it be on top level?
3. The page does not work, I couldn't use it.

abhigyank commented:
@robgev you can checkout https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages for commit guidilines and update the commit accordingly.

akashnimare commented:
> Sure thing, I will add snapcraft.yaml, but I don't exactly know where. Should it be on top level?

Should be in the `/snap/snapcraft.yaml`.

robgev committed:
Build: Add snapcraft.yaml file and gitignore the built files

This will add snapcraft.yaml file. Also, as crafting the snap
will add some build files in snap/ folder, those build files
were ignored.

robgev commented:
@akashnimare I added the snapcraft.yaml file to the directory, I don't think I can push the snap, I tried but it failed (probably because I don't have access).
Commit messages were fixed too.

One more thing, have you tried installing and running the snap? I got some error and will probably need some help.

Thanks.

akashnimare commented:
> One more thing, have you tried installing and running the snap? I got some error and will probably need some help.

Yeah, it works as expected. I get following errors but I think we can ignore this.
```
➜  dist git:(robgev-snap) zulip
Gtk-Message: Failed to load module "overlay-scrollbar"
Gtk-Message: Failed to load module "gail"
Gtk-Message: Failed to load module "atk-bridge"
Gtk-Message: Failed to load module "canberra-gtk-module"
```

robgev commented:
So should I fix or just ignore this @akashnimare? I think it will be just a few lines in snapcraft.yaml file.

akashnimare commented:
Fix it if you could otherwise it's fine.

robgev commented:
I will try till the end of today (like 5 hours from the moment of writing :) ). If I don't succeed or find any clue, then we will skip it. :)

akashnimare commented:
@robgev any updates on this?

robgev commented:
I think so, this errors are fixed when you install some libs with package manager. I am not sure that those libraries have the same name for all the distributions tho. What do you think should we merge it  or no?

akashnimare commented:
I'm getting the same errors with other snaps also so this could be the problem with the snap format. I run the snap locally and didn't notice any errors or something so we are good here. Merged. Thanks, @robgev.
